.github/workflows/main.yml

```
@@ -1,9 +1,12 @@
-name: CI
+name: CI Testing Pipeline
 
 on:
   push:
-    branches: [ "main" ]
+    # run for all main branch commits and on ci-* branches for testing new CI
+    # features.
+    branches: [ "main", "ci-*" ]
   pull_request:
+    # run on all pull requests against main.
```

```suggestion
    branches: [ "main", "ci-*" ]
  pull_request:
```
? These comments are a bit redundant and just add noise.

---

Ok I can remove those.